be/src/olap/memtable_flush_executor.cpp

```diff
@@ -35,15 +31,15 @@ std::ostream& operator<<(std::ostream& os, const FlushStatistic& stat) {
 }
 
 OLAPStatus FlushToken::submit(std::shared_ptr<MemTable> memtable) {
```

Comment:
why not use `const std::shared_ptr<MemTable>&` to avoid a copy assignment?

Reply:
After the `submit()` returns, the upper layer will immediately reset the `memtable` (let it be `nullptr`), and the lower thread pool may not yet start processing it. If a reference is used, when the thread pool starts to process this `memtable`, the corresponding object may have become invalid.

Reply:
`std::bind` below copies `memtable`, which will keep it valid in your mentioned situation. So I think it is OK to make the function's argument reference.

Reply:
Oh, I get you. I will change the param to a reference.